### PR TITLE
Oxidize recovery device

### LINKED ISSUE #3866: Oxidize `load_recovery_device`, `save_recovery_device`
Respect permission on creation:
 - Create (if needed) the base folder with permission `u=rwx,g=,o=`
 - The file must have the permission `u=rw,g=,o=`

> Would require to have to test to check for the permission (at least for the created file)

### PR DESCRIPTION
**Pull request by Volham22**
# What has changed ?

This PR closes  #3866.

- [ ] Does this PR affect the User ?

### CODE REVIEW

File: oxidation/libparsec/crates/client_types/src/local_device_file.rs [hunk 1]
```diff
@@ -291,7 +291,7 @@ pub fn list_available_devices(config_dir: &Path) -> LocalDeviceResult<Vec<Availa
     Ok(list)
 }
 
-pub async fn save_recovery_device(
```
[the remark this answers is not in the document]
Comment (Volham22) [reply oxidation/libparsec/crates/client_types/src/local_device_file.rs:294]: There's no asynchronous tasks are done in this function so I removed it. We can still do IO operations asynchronously.

[the remark this answers is not in the document]
Comment (TimeEngineer) [reply oxidation/libparsec/crates/client_types/src/local_device_file.rs:294]: cf https://github.com/Scille/parsec-cloud/pull/3892#discussion_r1067875373

[the remark this answers is not in the document]
Comment (Volham22) [reply oxidation/libparsec/crates/client_types/src/local_device_file.rs]: Done. I had to modify `FutureIntoCoroutine` a bit to make it work

[the remark this answers is not in the document]
Comment (FirelightFlagboy) [reply oxidation/libparsec/crates/client_types/src/local_device_file.rs]: for `FutureIntoCoroutine` take a look at #3913

[the remark this answers is not in the document]
Comment (Volham22) [reply oxidation/libparsec/crates/client_types/src/local_device_file.rs]: Should I wait #3913 to be merged and revert my changes for `FututreIntoCoroutine`?

[the remark this answers is not in the document]
Comment (FirelightFlagboy) [reply oxidation/libparsec/crates/client_types/src/local_device_file.rs]: It's merged ;)

File: src/runtime.rs [hunk 2]
```diff
@@ -83,16 +83,41 @@ impl TokioTaskAborterFromTrio {
 
 #[pyclass]
 pub(crate) struct FutureIntoCoroutine(
-    Option<Pin<Box<dyn Future<Output = PyResult<()>> + Send + 'static>>>,
+    Option<Pin<Box<dyn Future<Output = PyResult<PyObject>> + Send + 'static>>>,
 );
 
 impl FutureIntoCoroutine {
+    #[allow(dead_code)]
     pub fn new<F>(fut: F) -> Self
```
[the remark this answers is not in the document]
Comment (FirelightFlagboy) [reply src/runtime.rs:91]: I need a rebase for #3913 ;)

[the remark this answers is not in the document]
Comment (Volham22) [reply src/runtime.rs:91]: I'll remove this implementation and rebase with master. The issue I had has been fixed in #3913

File: oxidation/libparsec/crates/client_types/src/local_device_file.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (Volham22) [reply oxidation/libparsec/crates/client_types/src/local_device_file.rs]: Done.

**Review by FirelightFlagboy — APPROVED** [review]

**Pull request merged**